core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -151,6 +152,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                         final Throwable cause = nettyReq.decoderResult().cause();
                         if (cause instanceof TooLongHttpLineException) {
                             fail(id, null, HttpStatus.REQUEST_URI_TOO_LONG, "Too Long URI", cause);
+                        } else if (cause instanceof TooLongHttpHeaderException) {
```

Comment:
double checked that `TooLongHttpHeaderException` is not related to `TooLongHttpLineException`, so it doesn't matter which if statement is checked first 👍